### PR TITLE
 Fixed: imod_coupler couldn't handle the case where mf6 was configure…

### PR DESCRIPTION
**Pull request by mjr-deltares**
…d with the STORAGECOEFFICIENT option in the STO package.

### CODE REVIEW

**Comment by mjr-deltares** [comment]
for reference, this is the piece of code in MODFLOW 6:

`!> @brief Calculate the specific storage capacity
  !!
  !! Function to calculate the specific storage capacity using
  !! the cell geometry and the specific storage or storage coefficient.
  !!
  !! @return      sc1               specific storage capacity
  !<
  pure function SsCapacity(istor_coef, top, bot, area, ss) result(sc1)
    ! -- dummy variables
    integer(I4B), intent(in) :: istor_coef  !< flag indicating if ss is the storage coefficient
    real(DP), intent(in) :: top             !< top of cell
    real(DP), intent(in) :: bot             !< bottom of cell
    real(DP), intent(in) :: area            !< horizontal cell area
    real(DP), intent(in) :: ss              !< specific storage or storage coefficient
    ! -- local variables
    real(DP) :: sc1
    real(DP) :: thick
    ! -- calculate specific storage capacity
    if (istor_coef == 0) then
      thick = top - bot
    else
      thick = DONE
    end if
    sc1 = ss*thick*area
    !
    ! -- return
    return
  end function SsCapacity`

File: imod_coupler/drivers/metamod/metamod.py [hunk 1]
```diff
@@ -212,16 +215,24 @@ def couple(self) -> None:
...
+        # MetaSWAP gives SC1*area, MODFLOW by default needs SS, convert here.
+        # When MODFLOW is configured to use SC1 explicitly via the
+        # STORAGECOEFFICIENT option in the STO package, only the multiplication
+        # by area needs to be undone
+        conversion_terms = None
+        if self.mf6_has_sc1:
+            conversion_terms = 1.0 / self.mf6_area
+        else:
+            conversion_terms = 1.0 / np.multiply(
+                self.mf6_area, self.mf6_top - self.mf6_bot
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/metamod/metamod.py:227]:
```suggestion
            conversion_terms = 1.0 / self.mf6_area * (self.mf6_top - self.mf6_bot)
```

File: imod_coupler/drivers/metamod/metamod.py [hunk 2]
```diff
@@ -212,16 +215,24 @@ def couple(self) -> None:
...
-        # following the definition on specific storage in chapter 5 of
-        # the MODFLOW manual, but, this needs to be solved in MetaSWAP!!
-        sc1_to_ss = 1.0 / np.multiply(self.mf6_area, self.mf6_top - self.mf6_bot)
-        area_conversion = dia_matrix(
-            (sc1_to_ss, [0]),
+        # MetaSWAP gives SC1*area, MODFLOW by default needs SS, convert here.
+        # When MODFLOW is configured to use SC1 explicitly via the
+        # STORAGECOEFFICIENT option in the STO package, only the multiplication
+        # by area needs to be undone
+        conversion_terms = None
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/metamod/metamod.py:222]: Not strictly necessary here

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: imod_coupler/drivers/metamod/metamod.py (hunk 1 above)
Comment (mjr-deltares) [reply imod_coupler/drivers/metamod/metamod.py:227]: will change

**Review by Hofer-Julian — APPROVED** [review]
Will merge as soon as @JoerivanEngelen adapted the test case comparing specific storage with storage coefficient.

**Pull request merged**